test/application/widgets/theme/highlightable_text_test.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:memo/application/widgets/theme/highlightable_text.dart';
+
+import '../../../utils/widget_pump.dart';
+
+void main() {
+  const fakeText = 'A fake text';
+  const fakeTextStyle = TextStyle(color: Colors.white);
+  const fakeHighlight = 'fake';
+  const fakeHighlightStyle = TextStyle(color: Colors.red);
+
+  testWidgets('should not style highlight prefix', (tester) async {
```

Comment:
I think all of this tests could be merged into 2: `should use the default textStyle on anything other than the highlightable text` and `should use the highlightedStyle on highlighted text`.